MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -815,30 +695,32 @@ public final class MagazineLayout: UICollectionViewLayout {
       return
     }
 
-    if context.invalidateEverything {
-      prepareActions.formUnion([.recreateSectionModels, .lazilyCreateLayoutAttributes])
-    }
+    let shouldInvalidateLayoutMetrics = !context.invalidateEverything &&
+      !context.invalidateDataSourceCounts
```

Comment:
This was added to move this logic out of `prepare` and into here `invalidateLayout(with:)` https://github.com/airbnb/MagazineLayout/pull/94/files#diff-d77816f807be47f68fc24a016836021fbaa98c7aa4458d7b908a6fda34a6f4b1L111-L113

Reply:
I suspect that this won't even be needed once I remove the upfront section model creating from `prepare`, which is the next step after this PR.

Reply:
Reminder that `invalidateLayout` is called first (for a variety of reasons) then immediately after, `prepare` is called. We use `PrepareActions` to send some information about the invalidation reason from here to `prepare`.